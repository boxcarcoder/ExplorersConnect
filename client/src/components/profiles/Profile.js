import React, { Fragment, useEffect } from 'react';
import { Link } from 'react-router-dom';

import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getProfileById } from '../../actions/profile';

import Spinner from '../layout/Spinner';

const Profile = ({
  profileState: { profile, loading },
  getProfileById,
  match,
}) => {
  //when a user's profile page is first loaded, retrieve the profile and store it in the profile redux state
  useEffect(() => {
    getProfileById(match.params.id);
  }, [getProfileById, match.params.id]);

  //for async operation. need to wait for profile to be loaded into the profile redux state before proceeding
  if (profile === null || loading) {
    return (
      <Fragment>
        <Spinner />
      </Fragment>
    );
  }

  const {
    user: { name, avatar }, //this field countains a user field, which contains name and avatar
    bio,
    location,
    Hiking,
    Camping,
    Kayaking,
    Rafting,
    Skiing,
    Snowboarding,
    Rockclimbing,
    destinations,
    gears,
    social,
  } = profile;

  // ======== Check for favorite activities ========
  const checkHike = () => {
    if (Hiking) {
      return (
        <div className='p-1'>
          <i className='fas fa-campground'></i> Camping
        </div>
      );
    }
  };

  const checkCamp = () => {
    if (Camping) {
      return (
        <div className='p-1'>
          <i className='fas fa-hiking'></i> Hiking
        </div>
      );
    }
  };

  const checkKayak = () => {
    if (Kayaking) {
      return (
        <div className='p-1'>
          <i className='fas fa-water'></i> Kayaking
        </div>
      );
    }
  };

  const checkRaft = () => {
    if (Rafting) {
      return (
        <div className='p-1'>
          <i className='fas fa-tint'></i> Rafting
        </div>
      );
    }
  };

  const checkSki = () => {
    if (Skiing) {
      return (
        <div className='p-1'>
          <i className='fas fa-skiing'></i> Skiing
        </div>
      );
    }
  };

  const checkSnowboard = () => {
    if (Snowboarding) {
      return (
        <div className='p-1'>
          <i className='fas fa-snowboarding'></i> Snowboarding
        </div>
      );
    }
  };

  const checkRockclimb = () => {
    if (Rockclimbing) {
      return (
        <div className='p-1'>
          <i className='fas fa-mountain'></i> Rockclimbing
        </div>
      );
    }
  };

  // ======== Check for favorite destinations for each activity ========
  const checkHikeDestinations = () => {
    if (Hiking) {
      return (
        <Fragment>
          <p>
            <strong>Hike Trails</strong>
            <ul>
              {destinations.map((destination) => (
                <li>{destination.hikingTrails}</li>
              ))}
            </ul>
          </p>
          <div className='line'></div>
        </Fragment>
      );
    }
  };

  const checkCampDestinations = () => {
    if (Camping) {
      return (
        <Fragment>
          <p>
            <strong>Campsites</strong>
            <ul>
              {destinations.map((destination) => (
                <li>{destination.campSites}</li>
              ))}
            </ul>
          </p>
          <div className='line'></div>
        </Fragment>
      );
    }
  };

  const checkWaterDestinations = () => {
    if (Kayaking || Rafting) {
      return (
        <Fragment>
          <p>
            <strong>Lakes, Rivers, or Oceans</strong>
            <ul>
              {destinations.map((destination) => (
                <li>{destination.waterAreas}</li>
              ))}
            </ul>
          </p>
          <div className='line'></div>
        </Fragment>
      );
    }
  };

  const checkSnowDestinations = () => {
    if (Skiing || Snowboarding) {
      return (
        <Fragment>
          <p>
            <strong>Slopes</strong>
            <ul>
              {destinations.map((destination) => (
                <li>{destination.slopes}</li>
              ))}
            </ul>
          </p>
          <div className='line'></div>
        </Fragment>
      );
    }
  };

  const checkClimbDestinations = () => {
    if (Rockclimbing) {
      return (
        <Fragment>
          <p>
            <strong>Crags</strong>
            <ul>
              {destinations.map((destination) => (
                <li>{destination.crags}</li>
              ))}
            </ul>
          </p>
          <div className='line'></div>
        </Fragment>
      );
    }
  };

  // ======== Check for favorite gears for each activity ========
  const checkHikeGears = () => {
    if (Hiking) {
      return (
        <Fragment>
          <p>
            <strong>Hiking</strong>
            <ul>
              {gears.map((gear) => (
                <li>{gear.hikeGear}</li>
              ))}
            </ul>
          </p>
          <div className='line'></div>
        </Fragment>
      );
    }
  };

  const checkCampGears = () => {
    if (Camping) {
      return (
        <Fragment>
          <p>
            <strong>Camping</strong>
            <ul>
              {gears.map((gear) => (
                <li>{gear.campGear}</li>
              ))}
            </ul>
          </p>
          <div className='line'></div>
        </Fragment>
      );
    }
  };

  const checkWaterGears = () => {
    if (Kayaking || Rafting) {
      return (
        <Fragment>
          <p>
            <strong>Lakes, Rivers, or Oceans</strong>
            <ul>
              {gears.map((gear) => (
                <li>{gear.waterGear}</li>
              ))}
            </ul>
          </p>
          <div className='line'></div>
        </Fragment>
      );
    }
  };

  const checkSnowGears = () => {
    if (Skiing || Snowboarding) {
      return (
        <Fragment>
          <p>
            <strong>Slopes</strong>
            <ul>
              {gears.map((gear) => (
                <li>{gear.snowGear}</li>
              ))}
            </ul>
          </p>
          <div className='line'></div>
        </Fragment>
      );
    }
  };

  const checkClimbGears = () => {
    if (Rockclimbing) {
      return (
        <Fragment>
          <p>
            <strong>Rock Climbing</strong>
            <ul>
              {gears.map((gear) => (
                <li>{gear.rockClimbingGear}</li>
              ))}
            </ul>
          </p>
          <div className='line'></div>
        </Fragment>
      );
    }
  };

  // ======== Check for provided social media links ========
  const checkInstagram = () => {
    if (!social || !social.instagram) {
      return (
        <div className='invalid-icon'>
          <i className='fab fa-instagram fa-2x'></i>
        </div>
      );
    } else if (social.instagram) {
      return (
        <div className='valid-icon'>
          <a href={social.instagram} target='_blank' rel='noopener noreferrer'>
            <i className='fab fa-instagram fa-2x'></i>
          </a>
        </div>
      );
    }
  };

  const checkTwitter = () => {
    if (!social || !social.twitter) {
      return (
        <div className='invalid-icon'>
          <i className='fab fa-twitter fa-2x'></i>
        </div>
      );
    } else if (social.twitter) {
      return (
        <div className='valid-icon'>
          <a href={social.twitter} target='_blank' rel='noopener noreferrer'>
            <i className='fab fa-twitter fa-2x'></i>
          </a>
        </div>
      );
    }
  };

  const checkFacebook = () => {
    if (!social || !social.facebook) {
      return (
        <div className='invalid-icon'>
          <i className='fab fa-facebook fa-2x'></i>
        </div>
      );
    } else if (social.facebook) {
      return (
        <div className='valid-icon'>
          <a href={social.facebook} target='_blank' rel='noopener noreferrer'>
            <i className='fab fa-facebook fa-2x'></i>
          </a>
        </div>
      );
    }
  };

  const checkYoutube = () => {
    if (!social || !social.youtube) {
      return (
        <div className='invalid-icon'>
          <i className='fab fa-youtube fa-2x'></i>
        </div>
      );
    } else if (social.youtube) {
      return (
        <div className='valid-icon'>
          <a href={social.youtube} target='_blank' rel='noopener noreferrer'>
            <i className='fab fa-youtube fa-2x'></i>
          </a>
        </div>
      );
    }
  };

  const checkSocialMedia = () => {
    return (
      <Fragment>
        <div className='vert-m-1 icons-list'>
          {checkInstagram()}
          {checkTwitter()}
          {checkFacebook()}
          {checkYoutube()}
        </div>
      </Fragment>
    );
  };

  return (
    <Fragment>
      <Link to='/profiles' className='btn'>
        Back to Profiles
      </Link>
      <div className='vert-m-1 profile-grid'>
        {/* Top */}
        <div className='bg-primary p-2 profile-top'>
          <img className='curved-img vert-m-1' src={avatar} alt='profilePic' />

          <h1 className='large'>{name}</h1>
          <p>{location}</p>
          {checkSocialMedia()}
        </div>

        {/* Bio and Passions */}
        <div className='bg-light p-2 profile-about'>
          <h2 className='text-primary'>Bio</h2>
          <p>{bio}</p>

          <div className='line'></div>
          <h2 className='text-primary'>Passions</h2>
          <div className='passions'>
            {checkHike()}
            {checkCamp()}
            {checkKayak()}
            {checkRaft()}
            {checkSki()}
            {checkSnowboard()}
            {checkRockclimb()}
          </div>
        </div>

        {/* Destinations */}
        <div className='bg-light p-2 profile-location'>
          <h2 className='text-primary'>Explored</h2>
          {checkHikeDestinations()}
          {checkCampDestinations()}
          {checkWaterDestinations()}
          {checkSnowDestinations()}
          {checkClimbDestinations()}
        </div>

        {/* Gears */}
        <div className='bg-light p-2 profile-gear'>
          <h2 className='text-primary'>Gear</h2>
          {checkHikeGears()}
          {checkCampGears()}
          {checkWaterGears()}
          {checkSnowGears()}
          {checkClimbGears()}
        </div>
      </div>
    </Fragment>
  );
};

Profile.propTypes = {
  profileState: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
  profileState: state.profile,
});

export default connect(mapStateToProps, { getProfileById })(Profile);
