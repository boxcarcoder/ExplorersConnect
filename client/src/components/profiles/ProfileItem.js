import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';

import PropTypes from 'prop-types';

const ProfileItem = ({
  profile: {
    user: { _id, name, avatar },
    Hiking,
    Camping,
    Kayaking,
    Rafting,
    Skiing,
    Snowboarding,
    Rockclimbing,
    Other,
    location,
    bio
  }
}) => {

  const checkHike = () => {
    if (Hiking) {
      return (
        <li className='text-primary'>
          <i className='fas fa-check'></i> Hiking
        </li>
      );
    }
  };

  const checkCamp = () => {
    if (Camping) {
      return (
        <li className='text-primary'>
          <i className='fas fa-check'></i> Camping
        </li>
      );
    }
  };

  const checkKayak = () => {
    if (Kayaking) {
      return (
        <li className='text-primary'>
          <i className='fas fa-check'></i> Kayaking
        </li>
      );
    }
  };

  const checkRaft = () => {
    if (Rafting) {
      return (
        <li className='text-primary'>
          <i className='fas fa-check'></i> Rafting
        </li>
      );
    }
  };

  const checkSki = () => {
    if (Skiing) {
      return (
        <li className='text-primary'>
          <i className='fas fa-check'></i> Skiing
        </li>
      );
    }
  };

  const checkSnowboard = () => {
    if (Snowboarding) {
      return (
        <li className='text-primary'>
          <i className='fas fa-check'></i> Snowboarding
        </li>
      );
    }
  };

  const checkRockclimb = () => {
    if (Rockclimbing) {
      return (
        <li className='text-primary'>
          <i className='fas fa-check'></i> Rockclimbing
        </li>
      );
    }
  };

  return (
    <Fragment>
      <div className='profile bg-light'>
        <img className='round-img' src={avatar} alt='hiker' />
        <div className='m-3'>
          <h2>{name}</h2>
          <p>{location}</p>
          <p>{bio}</p>
          <Link to='/profile' className='btn btn-primary'>
            View Profile
          </Link>
        </div>
        <ul>
          {checkHike()}
          {checkCamp()}
          {checkKayak()}
          {checkRaft()}
          {checkSki()}
          {checkSnowboard()}
          {checkRockclimb()}
        </ul>
      </div>
    </Fragment>
  );
};

ProfileItem.propType = {
  profile: PropTypes.object.isRequired
};

export default ProfileItem;
