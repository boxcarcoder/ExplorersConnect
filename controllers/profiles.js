const profilesRouter = require('express').Router();
const mongoose = require('mongoose');
const Profile = require('../models/Profile');
const auth = require('../middlewares/auth');
const { check, validationResult } = require('express-validator');

// @route   GET api/profiles/me
// @desc    Fetch a user's profile
// @access  Private
profilesRouter.get('/me', auth, async (req, res) => {
  try {
    // find the profile of a requested user
    let profile = await Profile.findOne({ user: req.user.id });

    if (!profile) {
      res.status(400).json('This profile does not exist.');
    }

    res.json(profile);
  } catch (err) {
    console.error(err.message);
    res.status(500).json('Server error.');
  }
});

// @route   POST api/profiles
// @desc    Create a profile for a user
// @access  Private
profilesRouter.post(
  '/',
  [
    auth,
    [
      check('bio', 'Please provide a bio')
        .not()
        .isEmpty(),
      check('country', 'Please provide your current country')
        .not()
        .isEmpty()
    ]
  ],
  async (req, res) => {
    let errors = validationResult(req);
    if (!errors.isEmpty()) {
      return res.status(400).json({ errors: errors.array() });
    }

    // create a profile for a user
    const {
      bio,
      country,
      //passions,
      gears,
      youtube,
      twitter,
      facebook,
      instagram
    } = req.body;

    //Build object for the profile's fields
    let profileFields = {};
    profileFields.user = req.user.id;
    if (bio) profileFields.bio = bio;
    if (country) profileFields.country = country;
    if (gears) {
      profileFields.gears = gears.split(',').map(gear => gear.trim()); //turn comma separated string into array by split, then map to remove spaces from each index
    }

    //Build social object for our profile's fields object
    profileFields.social = {};
    if (youtube) profileFields.social.youtube = youtube;
    if (twitter) profileFields.social.twitter = twitter;
    if (facebook) profileFields.social.facebook = facebook;
    if (instagram) profileFields.social.instagram = instagram;

    try {
      let profile = await Profile.findOne({ user: req.user.id });

      if (profile) {
        //update the profile
        profile = await Profile.findOneAndUpdate(
          { user: req.user.id },
          { $set: profileFields },
          { new: true }
        );
        return res.json(profile);
      }

      // Create a new profile
      profile = new Profile(profileFields);
      await profile.save();
      res.json(profile);
    } catch (err) {
      console.error(err.message);
      res.status(500).json('Server error.');
    }
  }
);

module.exports = profilesRouter;

// if (passions) {
//   profileFields.passions = passions
//     .split(',')
//     .map(passion => passion.trim());
// }
