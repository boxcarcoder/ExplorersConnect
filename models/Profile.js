const mongoose = require('mongoose');

const profileSchema = new mongoose.Schema({
  user: {
    //every profile should be associated with a user
    type: mongoose.Schema.Types.ObjectId, //{_id:ObjectId:("..."), ...}
    ref: 'User' //reference the model we're referring to
  },
  bio: {
    type: String,
    required: true
  },
  location: {
    type: String,
    required: true
  },
  passions: [
    {
      hiking: {
        locations: {
          type: [String]
        },
        favorite: {
          type: Boolean,
          default: false
        }
      },
      camping: {
        locations: {
          type: [String]
        },
        favorite: {
          type: Boolean,
          default: false
        }
      },
      waterSports: {
        locations: {
          type: [String]
        },
        favorite: {
          type: Boolean,
          default: false
        }
      },
      snowSports: {
        locations: {
          type: [String]
        },
        favorite: {
          type: Boolean,
          default: false
        }
      },
      rockClimbing: {
        locations: {
          type: [String]
        },
        favorite: {
          type: Boolean,
          default: false
        }
      }
    }
  ],
  gears: [
    {
      hiking: {
        type: [String]
      },
      camping: {
        type: [String]
      },
      waterSports: {
        type: [String]
      },
      snowSports: {
        type: [String]
      },
      rockClimbing: {
        type: [String]
      }
    }
  ],
  social: {
    youtube: {
      type: String
    },
    twitter: {
      type: String
    },
    facebook: {
      type: String
    },
    instagram: {
      type: String
    }
  },
  date: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('Profile', profileSchema);
