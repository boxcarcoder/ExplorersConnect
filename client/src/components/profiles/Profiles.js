import React, { Fragment } from 'react';
import ProfileItem from './ProfileItem';
import Spinner from '../layout/Spinner';

//redux
import { connect } from 'react-redux';
import { getAllProfiles } from '../../actions/profile';
import PropTypes from 'prop-types';

export const Profiles = ({
  getAllProfiles,
  profileState: { profiles, loading },
}) => {
  //on the Profiles page's first load, retrieve all profiles and save it into the profile redux state
  React.useEffect(() => {
    getAllProfiles();
  }, []); //*** */

  if (!profiles.length || loading) {
    return <Spinner />;
  }
  return (
    <Fragment>
      <h1 className='large text-primary'>Explorers</h1>
      <p className='lead'>
        <i className='fab fa-connectdevelop'></i> Browse and connect with
        explorers
      </p>
      <div className='profiles'>
        {profiles.map((profile) => (
          <ProfileItem key={profile._id} profile={profile} />
        ))}
      </div>
    </Fragment>
  );
};

Profiles.propTypes = {
  getAllProfiles: PropTypes.func.isRequired,
  profileState: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
  profileState: state.profile,
});

export default connect(mapStateToProps, { getAllProfiles })(Profiles);
