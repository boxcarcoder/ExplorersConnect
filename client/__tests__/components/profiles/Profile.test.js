import React from 'react';
import { shallow } from 'enzyme';

// Import the non-connected component
import { Profile } from '../../../src/components/profiles/Profile';

// Globals for testing
let wrapper;
let props;
let mockGetProfileById = jest.fn();
const mockRouteParam = 1;


describe('<Profile /> component.', () => {
    describe('When the component is rendered,', () => {
        test('the getProfileById() action is fired.', () => {
            // Spy on the useEffect from React so we can redefine it.
            let useEffect = jest.spyOn(React, 'useEffect');

            const mockUseEffect = () => {
                useEffect.mockImplementationOnce((f) => f());
            };

            props = {
                profileState: {
                    profile: {
                        user: 'test user id',
                        bio: 'test user bio',
                        location: 'test user location',
                    },
                    loading: false            
                },
                getProfileById: mockGetProfileById,
                match: {
                    params: {
                        id: mockRouteParam
                    }
                }
            };

            // Execute the test.
            mockUseEffect()    
            wrapper = shallow(<Profile {...props} />);
            expect(mockGetProfileById).toHaveBeenCalled();
        });
    });

    describe('Successfully renders', () => {
        beforeEach(() => {
            props = {
                profileState: {
                    profile: {
                        user: 'test user id',
                        bio: 'test user bio',
                        location: 'test user location',
                        Hiking: true,
                        Camping: true,
                        Kayaking: true,
                        Rafting: true,
                        Skiing: true,
                        Snowboarding: true,
                        Rockclimbing: true,
                        social: {
                            instagram: 'test instagram link',
                            twitter: 'test twitter link',
                            facebook: 'test facebook link',
                            youtube: 'test youtube link',
                        },
                        destinations: [
                            {
                                hikingTrails: 'test hike trail',
                                campSites: 'test camp site',
                                waterAreas: 'test water area',
                                slopes: 'test slope',
                                crags: 'test cliff',
                            }                            
                        ],
                        gears: [
                            {
                                hikeGear: 'test hike gear',
                                campGear: 'test camp gear',
                                waterGear: 'test water gear',
                                snowGear: 'test snow gear',
                                rockClimbingGear: 'test rock gear',

                            },

                        ]
                    },
                    loading: false,

                },
                getProfileById: mockGetProfileById,
                match: {
                    params: {
                        id: mockRouteParam
                    }
                }
            };

            wrapper = shallow(<Profile {...props} />);    
        });

        test('the Profile component.', () => {
            expect(wrapper).toMatchSnapshot();
        });

        test('a spinner while the loading state in the Profile redux state is true.', () => {
            // Set loading to true
            props = {
                profileState: {
                    profile: {
                        user: 'test user id',
                        bio: 'test user bio',
                        location: 'test user location',
                        Hiking: true,
                        Camping: true,
                        Kayaking: true,
                        Rafting: true,
                        Skiing: true,
                        Snowboarding: true,
                        Rockclimbing: true,
                        social: {
                            instagram: 'test instagram link',
                            twitter: 'test twitter link',
                            facebook: 'test facebook link',
                            youtube: 'test youtube link',
                        },
                        destinations: [
                            {
                                hikingTrails: 'test hike trail',
                                campSites: 'test camp site',
                                waterAreas: 'test water area',
                                slopes: 'test slope',
                                crags: 'test cliff',
                            }                            
                        ],
                        gears: [
                            {
                                hikeGear: 'test hike gear',
                                campGear: 'test camp gear',
                                waterGear: 'test water gear',
                                snowGear: 'test snow gear',
                                rockClimbingGear: 'test rock gear',

                            },

                        ]
                    },
                    loading: true,
                },
                getProfileById: mockGetProfileById,
                match: {
                    params: {
                        id: mockRouteParam
                    }
                }
            };

            wrapper = shallow(<Profile {...props} />);    

            expect(wrapper.find('Spinner').exists()).toBe(true);
        });

        test('a spinner while the loading state the profile in Profile state is null.', () => {
            // Set profile data to null to true
            props = {
                profileState: {
                    profile: null,
                    loading: false,
                },
                getProfileById: mockGetProfileById,
                match: {
                    params: {
                        id: mockRouteParam
                    }
                }
            };

            wrapper = shallow(<Profile {...props} />);    

            expect(wrapper.find('Spinner').exists()).toBe(true);
        });

        test('a link to the profiles page.', () => {
            const link = wrapper.find('Link');
            expect(link.props().to).toBe('/profiles');
        });

        describe('an icon for', () => {
            test('hiking if hiking is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('div').at(10).find('i').hasClass('fa-hiking')).toBe(true);
            });

            test('camping if camping is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('div').at(11).find('i').hasClass('fa-campground')).toBe(true);
            });

            test('kayaking if kayaking is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('div').at(12).find('i').hasClass('fa-water')).toBe(true);
            });

            test('rafting if rafting is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('div').at(13).find('i').hasClass('fa-tint')).toBe(true);
            });

            test('skiing if skiing is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('div').at(14).find('i').hasClass('fa-skiing')).toBe(true);
            });

            test('snowboarding if snowboarding is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('div').at(15).find('i').hasClass('fa-snowboarding')).toBe(true);
            });

            test('rock climbing if rock climbing is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('div').at(16).find('i').hasClass('fa-mountain')).toBe(true);
            });
        });

        describe('a list of favorite destinations for', () => {
            test('hiking if hiking is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('p').at(2).find('li').exists()).toBe(true);
            });

            test('camping if camping is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('p').at(3).find('li').exists()).toBe(true);
            });

            test('kayaking if kayaking is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('p').at(4).find('li').exists()).toBe(true);
            });

            test('rafting if rafting is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('p').at(4).find('li').exists()).toBe(true);
            });

            test('skiing if skiing is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('p').at(5).find('li').exists()).toBe(true);
            });

            test('snowboarding if snowboarding is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('p').at(5).find('li').exists()).toBe(true);
            });

            test('rockClimbing if rock climbing is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('p').at(6).find('li').exists()).toBe(true);
            });
        });

        describe('a list of favorite gears for', () => { 
            test('hiking if hiking is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('p').at(7).find('li').exists()).toBe(true);
            });

            test('camping if camping is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('p').at(8).find('li').exists()).toBe(true);
            });

            test('kayaking if kayaking is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('p').at(9).find('li').exists()).toBe(true);
            });

            test('rafting if rafting is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('p').at(9).find('li').exists()).toBe(true);
            });

            test('skiing if skiing is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('p').at(10).find('li').exists()).toBe(true);
            });

            test('snowboarding if snowboarding is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('p').at(10).find('li').exists()).toBe(true);
            });

            test('rockClimbing if rock climbing is one of the profile\'s favorite activities.', () => {
                expect(wrapper.find('p').at(11).find('li').exists()).toBe(true);
            });
        });

        describe('a link to', () => { 
            test('instagram if the profile has an instagram link.', () => {
                expect(wrapper.find('div').at(3).find('a').props().href).toBe('test instagram link');
            });
    
            test('twitter if the profile has a twitter link.', () => {
                expect(wrapper.find('div').at(4).find('a').props().href).toBe('test twitter link'); 
            });
    
            test('facebook if the profile has a facebook link.', () => {
                expect(wrapper.find('div').at(5).find('a').props().href).toBe('test facebook link');
            });
    
            test('youtube if the profile has a youtube link.', () => {
                expect(wrapper.find('div').at(6).find('a').props().href).toBe('test youtube link');       
            });
        });    
    });  
});
