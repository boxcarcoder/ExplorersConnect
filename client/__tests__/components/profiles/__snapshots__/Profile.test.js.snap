// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> component. Successfully renders the Profile component. 1`] = `
<Fragment>
  <Link
    className="btn"
    to="/profiles"
  >
    Back to Profiles
  </Link>
  <div
    className="vert-m-1 profile-grid"
  >
    <div
      className="bg-primary p-2 profile-top"
    >
      <img
        alt="profilePic"
        className="curved-img vert-m-1"
      />
      <h1
        className="large"
      />
      <p>
        test user location
      </p>
      <div
        className="vert-m-1 icons-list"
      >
        <div
          className="valid-icon"
        >
          <a
            href="test instagram link"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              className="fab fa-instagram fa-2x"
            />
          </a>
        </div>
        <div
          className="valid-icon"
        >
          <a
            href="test twitter link"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              className="fab fa-twitter fa-2x"
            />
          </a>
        </div>
        <div
          className="valid-icon"
        >
          <a
            href="test facebook link"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              className="fab fa-facebook fa-2x"
            />
          </a>
        </div>
        <div
          className="valid-icon"
        >
          <a
            href="test youtube link"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              className="fab fa-youtube fa-2x"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      className="bg-light p-2 profile-about"
    >
      <h2
        className="text-primary"
      >
        Bio
      </h2>
      <p>
        test user bio
      </p>
      <div
        className="line"
      />
      <h2
        className="text-primary"
      >
        Passions
      </h2>
      <div
        className="passions"
      >
        <div
          className="p-1"
        >
          <i
            className="fas fa-hiking"
          />
           Hiking
        </div>
        <div
          className="p-1"
        >
          <i
            className="fas fa-campground"
          />
           Camping
        </div>
        <div
          className="p-1"
        >
          <i
            className="fas fa-water"
          />
           Kayaking
        </div>
        <div
          className="p-1"
        >
          <i
            className="fas fa-tint"
          />
           Rafting
        </div>
        <div
          className="p-1"
        >
          <i
            className="fas fa-skiing"
          />
           Skiing
        </div>
        <div
          className="p-1"
        >
          <i
            className="fas fa-snowboarding"
          />
           Snowboarding
        </div>
        <div
          className="p-1"
        >
          <i
            className="fas fa-mountain"
          />
           Rockclimbing
        </div>
      </div>
    </div>
    <div
      className="bg-light p-2 profile-location"
    >
      <h2
        className="text-primary"
      >
        Explored
      </h2>
      <p>
        <strong>
          Hike Trails
        </strong>
        <ul>
          <li>
            test hike trail
          </li>
        </ul>
      </p>
      <div
        className="line"
      />
      <p>
        <strong>
          Campsites
        </strong>
        <ul>
          <li>
            test camp site
          </li>
        </ul>
      </p>
      <div
        className="line"
      />
      <p>
        <strong>
          Lakes, Rivers, or Oceans
        </strong>
        <ul>
          <li>
            test water area
          </li>
        </ul>
      </p>
      <div
        className="line"
      />
      <p>
        <strong>
          Slopes
        </strong>
        <ul>
          <li>
            test slope
          </li>
        </ul>
      </p>
      <div
        className="line"
      />
      <p>
        <strong>
          Crags
        </strong>
        <ul>
          <li>
            test cliff
          </li>
        </ul>
      </p>
      <div
        className="line"
      />
    </div>
    <div
      className="bg-light p-2 profile-gear"
    >
      <h2
        className="text-primary"
      >
        Gear
      </h2>
      <p>
        <strong>
          Hiking
        </strong>
        <ul>
          <li>
            test hike gear
          </li>
        </ul>
      </p>
      <div
        className="line"
      />
      <p>
        <strong>
          Camping
        </strong>
        <ul>
          <li>
            test camp gear
          </li>
        </ul>
      </p>
      <div
        className="line"
      />
      <p>
        <strong>
          Lakes, Rivers, or Oceans
        </strong>
        <ul>
          <li>
            test water gear
          </li>
        </ul>
      </p>
      <div
        className="line"
      />
      <p>
        <strong>
          Slopes
        </strong>
        <ul>
          <li>
            test snow gear
          </li>
        </ul>
      </p>
      <div
        className="line"
      />
      <p>
        <strong>
          Rock Climbing
        </strong>
        <ul>
          <li>
            test rock gear
          </li>
        </ul>
      </p>
      <div
        className="line"
      />
    </div>
  </div>
</Fragment>
`;
